source/MapDetailPanel.cpp

```diff
@@ -589,6 +614,31 @@ void MapDetailPanel::DrawInfo()
 		uiPoint.Y() += 20.;
 	}
 
+	autoBuyY = uiPoint.Y();
+
+	// "Buy best" button goes after trade prices
+	if(player.CanTrade())
+	{
+		int canBuyBest = player.HasBestTrade(selectedSystem);
+		if(canBuyBest || player.Cargo().CommoditiesSize())
+		{
+			const Sprite *buyAllSprite = SpriteSet::Get("ui/planet dialog button"); //160x60
+			const auto alignCenter = Layout(160, Alignment::CENTER);
+			SpriteShader::Draw(buyAllSprite, uiPoint + Point(70, 35));
+			if(canBuyBest == 0)
+			{
+				font.Draw({"Sell Commodi_ties", alignCenter}, uiPoint + Point(-10, 28), medium);
+			}
+			else
+			{
+				font.Draw({canBuyBest == 2 ? "Fill Ship with" : "Fill Fleet with", alignCenter}, uiPoint + Point(-10, 20), medium);
```

Comment:
```suggestion
				font.Draw({canBuyBest == 2 ? "Fill Flagship with" : "Fill Fleet with", alignCenter}, uiPoint + Point(-10, 20), medium);
```
If it's going to fill the flagship just say so ^^

Reply:
Does the game often refer to your ship as "flagship" or maybe just "your ship"? This is also the button when you're the *only* ship, so 'flagship' would be inaccurate then. So this could be "your ship" I guess.